saleor/webhook/subscription_payload.py

```diff
@@ -0,0 +1,116 @@
+from typing import Any, Dict, Optional
+
+from celery.utils.log import get_task_logger
+from django.core.handlers.base import BaseHandler
+from django.http import HttpRequest
+from django.test.client import RequestFactory
+from graphene_django.settings import graphene_settings
+from graphene_django.views import instantiate_middleware
+from graphql import get_default_backend, parse
+from graphql.error import GraphQLSyntaxError
+from graphql.execution.middleware import MiddlewareManager
+from promise import Promise
+
+from ..app.models import App
+
+logger = get_task_logger(__name__)
+
+
+def validate_subscription_query(query: str) -> bool:
+    from ..graphql.api import schema
+
+    graphql_backend = get_default_backend()
+    try:
+        graphql_backend.document_from_string(schema, query)
+    except (ValueError, GraphQLSyntaxError):
+        return False
+    return True
+
+
+def initialize_context() -> HttpRequest:
+    """Prepare a request object for webhook subscription.
+
+    It creates a dummy request object and initialize middleware on it. It is required
+    to process a request in the same way as API logic does.
+    return: HttpRequest
+    """
+    handler = BaseHandler()
+    context = RequestFactory().request()
+    handler.load_middleware()
```

Comment:
Should we load middleware for generating payloads?

Reply:
We don't use middleware, it's expensive.

Reply:
All our resolvers assume that Django middlewares have been called and we have access to data like `request.discounts`, `request.plugins` etc. We load middlewares here to have the same input resolver data as we have from API calls. The second way would be manually attaching all dependencies that are added by middlewares.

Reply:
@fowczarek django